constraint-solver/src/effect.rs

```diff
@@ -25,6 +25,15 @@ pub enum Effect<T: FieldElement, V> {
     },
 }
 
+impl<T: FieldElement, V> Effect<T, V> {
+    pub fn from_range_constraint_update(variable: V, range_constraint: RangeConstraint<T>) -> Self {
+        match range_constraint.try_to_single_value() {
+            Some(value) => Effect::Assignment(variable, SymbolicExpression::Concrete(value)),
+            None => Effect::RangeConstraint(variable, range_constraint),
```

Comment:
We should handle Effect::RangeConstraint with a single value in the same way as Effect::Assignment, so I think this conversion is not needed.

Reason behind: If we already have a range constraint `[2-5]` and we get a new range constraint `[5-8]`, the combined range constraint is just `5`, but we only see that when we apply the range constraint update.